stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/plugin/state/channel/internal/ChannelMutableState.kt

```diff
@@ -83,6 +83,11 @@ internal class ChannelMutableState(
     /** the data to hide messages before */
     var hideMessagesBefore: Date? = null
 
+    /**
+     * Messages to show on the channels list when the state is inside search to show the latest message in the preview.
+     */
+    internal val cachedLatestMessages: MutableStateFlow<Map<String, Message>> = MutableStateFlow(emptyMap())
```

Comment:
Sometimes when you navigate to an older message via a quoted reply and then go back, the preview message of the channel you were in when you navigated to said message is different from the latest one and the channel is misplaced in the list (Affects both XML and Compose). This happens even when you are not in the search state.

I wonder if this could be the cause of it.

Reply:
Could be, will investigate this.

Reply:
Fixed the logic